docusaurus/docs/quickstarts/network/hosted/index.md

```diff
@@ -1,168 +1,258 @@
+
+### Firewall
+
+The first issue you will need to deal with is opening some ports. A network will consist of at least one controller and 
+at least one edge router. Both of these components will require ports to be open. For the controller you will need to 
+open a range of ports through your firewall:
+
+- `8440/tcp`: Edge Controller providing router control plane
+- `8441/tcp`: Edge Controller providing client sessions
+- `8442/tcp`: Edge Router providing client connections
+- `8443/tcp`: Ziti Admin Console (ZAC) [optional]
+
+These are the arbitrary ports we'll use in this example for convenience when specifying the firewall exception as a port range.
+
+## Express Install
+
+With the firewall ports open you will now be able to source the script which provides the expressInstall function and
+execute it. You probably should consider DNS names before actually running the script. By default, the script will use the output of the
+`hostname -s` command to determine a reasonable default for the PKI that will be generated as well as the
+configuration files that are output.  The script allows you to configure these values and if you are deploying to a cloud
+provider (AWS, Azure, OCI, IBM, Digital Ocean, GCP, etc.) you will almost certainly want to use the **public** DNS name
+of your instance. You possibly  want to override the IP address to use as well, but using DNS names is superior to using
+IP addresses in case your IP changes.
+
+The quickest and easiest thing to do, is simply find your external DNS name and set it into the EXTERNAL_DNS environment
+variable. For example:
+
+```bash
+export EXTERNAL_DNS="acme.example.com"
+```
+
+A random password for user "admin" will be generated and stored in the `*.env` file inside the express install home directory so that it can be easily referenced by the `zitiLogin` function. Optionally, you may override with a preferred password by setting `ZITI_PWD` now.
+
+```bash
+read -s ZITI_PWD
+# type or paste the password (not echoed) and press <enter>
+```
+
+:::note
+Make sure you have `jq` installed on your machine. From Debian distros that would look like:
+
+```bash
+sudo apt update && sudo apt install jq -y
+```
+
+:::
+
+Once you do that, you'll be able to
+execute these commands just as
+shown to have your
+controller and
+first edge router
+configured and ready to turn on:
+
+```bash
+# run me
+export EXTERNAL_IP="$(curl -s eth0.me)"       
+export ZITI_EDGE_CONTROLLER_IP_OVERRIDE="${EXTERNAL_IP}"
+export ZITI_EDGE_ROUTER_IP_OVERRIDE="${EXTERNAL_IP}"
+export ZITI_EDGE_CONTROLLER_HOSTNAME="${EXTERNAL_DNS}"
+export ZITI_EDGE_ROUTER_HOSTNAME="${EXTERNAL_DNS}"
+export ZITI_CTRL_PORT=8440
+export ZITI_EDGE_CONTROLLER_PORT=8441
+export ZITI_EDGE_ROUTER_PORT=8442
+
+# now download, source, and execute the expressInstall function
+source <(wget -qO- https://raw.githubusercontent.com/openziti/ziti/main/quickstart/docker/image/ziti-cli-functions.sh)
```

Comment:
Isn't this going to break for Bash 3 again? I thought that was the reason for `source /dev/stdin <<< ...`

Reply:
good catch, geoff. definitely don't want to change the way that gets sourced. on mac's the bash is ancient and this was necessary

Reply:
I too have found the `/dev/stdin` approach was necessary for default BASH 3 that shipped with latest macOS in 2013. Haven't checked since then. I guessed wrong that this was aimed at Linux servers. I'll have to redress a lot of assumptions, e.g. portable vs modern BASHisms. This is pretty bad news for the script.

Reply:
Zsh is the default since macOS Catalina (2019) and so most macOS users will be able to use modern shell-isms.

Reply:
We need a preamble near the top of this quickstart like "This is intended for Linux servers" or "This quickstart should work with modern macOS or Linux or WSL". I think this particular hosted-anywhere quickstart lends itself to the Linux server audience and I imagine the WSL and macOS crowds would use one of the local quickstarts instead.